Android/app/src/go/intra/split/direct_split.go

```diff
@@ -54,13 +54,20 @@ func (s *splitter) Write(b []byte) (int, error) {
 
 	// Setting `used` to true ensures that this code only runs once per socket.
 	s.used = true
-	b1, b2 := splitHello(b)
-	n1, err := conn.Write(b1)
-	if err != nil {
-		return n1, err
+	pkts, _ := splitHello(b)
+
+	// We did not use pkts.WriteTo(conn), because under the hood, the connection
+	// will use writev system call to write buffers, and writev may combine these
+	// buffers into one single write
+	n := 0
+	for _, pkt := range pkts {
```

Comment:
As discussed on chat, let's go back to only two TCP segments, so there are less things changing and it's easier to understand the impact.

Reply:
done